File: frontend/miniapp/src/api/couple.ts
import { request } from '@/utils/request'

export interface CoupleInfo {
  id: string
  partner_id: string
  partner_nickname: string
  partner_username: string | null
  partner_avatar: string | null
  partner_birthday: string | null
  partner_gender: string | null
  start_date: string
  days_together: number
  monthly_budget: number | null
}

export const coupleApi = {
  generate: () =>
    request<{ code: string; expires_in: number }>({ url: '/api/couple/generate', method: 'POST' }),

  confirm: (code: string, start_date?: string) =>
    request<{ message: string; couple_id: string }>({
      url: '/api/couple/confirm', method: 'POST', data: { code, start_date },
    }),

  info: () =>
    request<CoupleInfo>({ url: '/api/couple/info' }),

  unbind: () =>
    request<{ message: string }>({ url: '/api/couple/unbind', method: 'POST' }),

  updateBudget: (monthly_budget: number, month?: string) =>
    request<CoupleInfo>({ url: '/api/couple/budget', method: 'PUT', data: { monthly_budget, month } }),
}

File: backend/schemas/couple.py
from pydantic import BaseModel, Field
from datetime import date as Date
from typing import Optional


class GenerateCodeResponse(BaseModel):
    code: str
    expires_in: int = 300


class ConfirmPairRequest(BaseModel):
    code: str = Field(..., min_length=6, max_length=6)
    start_date: Optional[Date] = None


class CoupleInfo(BaseModel):
    id: str
    partner_id: str
    partner_nickname: str
    partner_username: Optional[str]
    partner_avatar: Optional[str]
    partner_birthday: Optional[Date]
    partner_gender: Optional[str]
    start_date: Date
    days_together: int
    monthly_budget: Optional[float] = None


class BudgetUpdate(BaseModel):
    monthly_budget: float = Field(..., gt=0)
    month: Optional[str] = None  # 格式 "2026-06"，不传则设置全局默认
